diff --git a/devEcmcSup/motor/ecmcMotorRecordAxis.cpp b/devEcmcSup/motor/ecmcMotorRecordAxis.cpp
index a6834a4f..e7b92c2f 100644
--- a/devEcmcSup/motor/ecmcMotorRecordAxis.cpp
+++ b/devEcmcSup/motor/ecmcMotorRecordAxis.cpp
@@ -717,15 +717,19 @@ asynStatus ecmcMotorRecordAxis::syncMotorSoftLimits(bool force, bool updateParam
   if (ecmcRTMutex)epicsMutexUnlock(ecmcRTMutex);
 
   if((enabledBwd == 0 && enabledFwd == 0) || (fValueBwd == 0 && fValueFwd == 0)) {
+#ifdef motorHighLimitROString
     asynMotorAxis::setDoubleParam(pC_->motorLowLimitRO_,  0);
     asynMotorAxis::setDoubleParam(pC_->motorHighLimitRO_, 0);
+#endif // ifdef motorHighLimitROString
     if (force) {
       asynMotorAxis::setDoubleParam(pC_->motorLowLimit_,  0);
       asynMotorAxis::setDoubleParam(pC_->motorHighLimit_, 0);
     }
   } else {
+#ifdef motorHighLimitROString
     asynMotorAxis::setDoubleParam(pC_->motorLowLimitRO_,  fValueBwd);
     asynMotorAxis::setDoubleParam(pC_->motorHighLimitRO_, fValueFwd);
+#endif // ifdef motorHighLimitROString
     if (force) {
       asynMotorAxis::setDoubleParam(pC_->motorLowLimit_,  fValueBwd);
       asynMotorAxis::setDoubleParam(pC_->motorHighLimit_, fValueFwd);
@@ -796,10 +800,14 @@ asynStatus ecmcMotorRecordAxis::readBackSoftLimits(bool updateMotor) {
   pC_->setIntegerParam(axisNo_, pC_->ecmcMotorRecordCfgDHLM_En_, enabledFwd);
   pC_->setDoubleParam(axisNo_, pC_->ecmcMotorRecordCfgDHLM_, fValueFwd);
 
+#ifdef motorHighLimitROString
   if(updateMotor) {
     pC_->setDoubleParam(axisNo_, pC_->motorLowLimitRO_, fValueBwd);
     pC_->setDoubleParam(axisNo_, pC_->motorHighLimitRO_, fValueFwd);
   }
+#else // ifdef motorHighLimitROString
+  (void)updateMotor;
+#endif // ifdef motorHighLimitROString
   
   pC_->callParamCallbacks();
 
